dev-requirements.txt

```diff
@@ -0,0 +1,3 @@
+pytest
+mypy
+pre-commit
```

Comment:
Should we also add flake8/ufmt/black/usort to dev requirements?

Reply:
it should come from precommit right?

Reply:
Yeah pre-commit will run these regardless, I am thinking more if developers want to run the checks themselves. But no strong preference here

Reply:
pre-commit will run more than just linting and autoformatting including license header, AST and etc.
If the user wants to run linting/formatter separately, they will need to install flake8/umft/black/usort themselves.
The versions are also important to ensure consistency with fb internal. We documented the versions in contributing guide some time ago.

Reply:
just added back the install instructions for ufmt etc in contributing